adr/0016-compliance-in-pepr.md

```diff
@@ -0,0 +1,51 @@
+# 14. Compliance in Pepr 
+
+Date: 2025-02-13
+
+
+## Status
+
+Pending
+
+
+## Context
+
+Pepr needs to generate reports for compliance purposes, demonstrating adherence to required security and compliance standards through specific bindings. To achieve this, it’s essential to map compliance controls to Pepr’s bindings. During our discovery process, we determined that Pepr’s role in compliance should focus on reporting rather than enforcement. This is because compliance controls are already enforced through the bindings themselves, and Pepr’s primary function is to report on the compliance status of resources.
+
+## Decision
+
+Extend the fluent API with a new keyword function that accepts a strongly typed control, allowing it to be associated with a given binding. This enables users to map compliance controls to Pepr’s bindings, facilitating the generation of compliance reports based on control status.
```

Comment:
Is the intent to stick to admission or provide a separate example/implementation? 

I believe admission enables compliance - but is not indicative of compliance as a whole.

`OnSchedule` I think has a lot of room for use here as the general baseline. 

Not to say this mechanism couldn't be used on admission - considering the levels of fidelity in doing so is important though.

Reply:
The answer influences some questions I had below on if module authors _need_ to update their modules or if this is outright a separate implementation.

Reply:
Good points, in this case, it is the module author that is writing the Pepr bindings that is responsible for enforcing the compliance. The job is Pepr Core is to enable reporting on the bindings and functions